### DISCUSSION TITLE
Hot Reload instead of SwiftUI Previews

### DISCUSSION THREAD

**Discussion by wojciech-kulik** [wojciech-kulik/xcodebuild.nvim#38] [Tips & Tricks]
As far as I know, it is not possible to show SwiftUI previews outside of Xcode. It is a built-in feature of Xcode.

However, you can workaround this problem by adding hot reload to your project. Check out these tools:
- https://github.com/johnno1962/InjectionIII
- https://github.com/krzysztofzablocki/Inject

**Comment by Almaz5200** [comment]
Does it work for you out of the box? I change targets frequently because of the nature of my project, and I find that I need to build needed target with Xcode before Injection picks up

**Reply by wojciech-kulik** [reply]
Hmm, I usually work with one target. Did you set the required build flags for all your targets?

**Reply by Almaz5200** [reply]
Yep, it does work when I build these targets with Xcode instead, and it continues to work even with Neovim afterwards. I'll investigate further and let you know what I find. Unfortunately, I have not yet been able to test it on a new version that includes logs in dap-ui. Perhaps that will provide some ideas on why this is happening.

**Reply by wojciech-kulik** [reply]
Keep in my that to see NSLog you should run the app without debugger (using xcodebuild.nvim commands). Otherwise, those entries won't appear. Not sure where InjectionIII prints logs. Hopefully, you will find something. Maybe Xcode triggers some extra steps required by the tool like indexing or something.

**Reply by Almaz5200** [reply]
So, I've finally been able to test it
I am not sure what changed between the versions, but updating both InjectionIII app and Injection package did the trick and the problem is gone now. My previous package version was 1.0.5, now it is 1.3.0
Before updating I was getting this error:

<img width="905" alt="Screenshot 2024-02-19 at 12 43 56" src="https://github.com/wojciech-kulik/xcodebuild.nvim/assets/21291840/203853f3-d6b6-47fe-bfef-1ee9d3c0f12f">

Thanks for your assistance!

**Comment by wojciech-kulik** [comment]
Moved to Wiki
